rj_gameplay/rj_gameplay/tactic/goalie_tactic.py

```diff
@@ -103,7 +105,82 @@ def get_requests(self, world_state: rc.WorldState,
         :return: A list of role requests for move skills needed
         """
 
+        def track_utility() -> float:
+            """
+            :return: the utility of track action
+            """
+            if ball_speed == 0:
+                ball_to_goal_time = 100
+            else:
+                ball_to_goal_time = ball_dist / ball_speed
+            return ball_to_goal_time / 100
+
+        def receive_utility() -> float:
+            """
+            :return: the utility of receive action
+            """
+            # this probably makes sense to stay hard coded like this
+            # but maybe there is a better option?
+            if ball_speed < .5 and (
```

Comment:
None of the utility() functions have params passed in, it would be better to pass them in explicitly. And for now hardcoded is fine, we can tune later.

Reply:
Since it's an inner function, it can access those params (I think). So should I make all of these functions separate from get_requests?

Reply:
Ahh, I didn't notice that. My bad. It's fine as is then.